python/mxnet/optimizer.py

```diff
@@ -379,7 +444,10 @@ def create_state(self, index, weight):
             momentum = zeros(weight.shape, weight.context, dtype=weight.dtype, stype=weight.stype)
         return momentum
 
-    def update(self, index, weight, grad, state):
+    def create_state(self, index, weight):
```

Comment:
There should be a _create_state_impl function that both create_mp_state and create_state use. And create_state should keep the original behavior (always multi_precision=False)

Reply:
I don't think _create_state_impl is necessary, since this is basically create_state.